src/types-and-traits.rst

```diff
@@ -2778,7 +2778,35 @@ Lifetime Elision
 .. rubric:: Legality Rules
 
 :dp:`fls_9wtuclhm7yz5`
-:t:`Lifetime elision` is a set of relaxations on the use of :t:`[lifetime]s`.
+:t:`Lifetime elision` is a set of rules that automatically insert :t:`[lifetime
+parameter]s` and/or :t:`[lifetime argument]s` when they are elided in the source
+code.
+
+:dp:`fls_JmP6O9zj8fkV`
+A :t:`lifetime` may be elided either implicitly or explicitly.
+
+:dp:`fls_5ZAQ9p7jQuc2`
+A :t:`lifetime` is elided explicitly if it is the ``'_`` :t:`lifetime`.
+
+:dp:`fls_YmUQ8ZiQuycp`
+A :t:`lifetime` is elided implicitly if it is absent.
+
+:dp:`fls_sIMN6Sd8xUZz`
+:t:`Lifetime elision` rules are introduced by certain :t:`[construct]s` and may
+be nested.
+
+:dp:`fls_dIyisjNIx9dC`
+If an elided :t:`lifetime` is encountered, the innermost set of :t:`lifetime
```

Comment:
What/which are these innermost (vs outermost?) set of rules?

Reply:
Ouf, I see what you mean. How about something long the lines of

An elided lifetime is subject to the lifetime elision rules of the narrowest context where the lifetime appears.

Or something along those lines. It is still somewhat unclear what "narrowest" means. How about

An elided lifetime is subject to the lifetime elision rules of the innermost construct where the lifetime appears.

I also don't like "appears". Any improvements on these suggestions?

Reply:
Okay, I've rewritten this sentence to hopefully be clearer